### PR TITLE
Add Prometheus statistics and an example to showcase them using Grafana

### PR DESCRIPTION
**Pull request by nigeldeakin**
This PR takes TJ's code changes in https://github.com/fnproject/fn/pull/357 and fixes a TODO in that code so that the `promhttp.Handler()` is created just once.  

I've also created an  example `examples/grafana` which shows how to run Prometheus and Grafana and install a dashboard `fn_grafana_dashboard.json` which showcases these metrics (queued/running/completed/failed) in a similar format to the `ui` tool. See screenshot....

grafanadashboard

### CODE REVIEW

**Comment by nigeldeakin** [comment]
Added screenshot. See
https://github.com/fnproject/fn/tree/add_prometheus_metrics/examples/grafana

File: examples/grafana/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Display runtime metrics using Prometheus and Grafana
+
+The Fn server exports metrics using [Prometheus](https://prometheus.io/). This allows [Grafana](https://grafana.com/) to be used to display these metrics graphically. 
+
+<img src="../../docs/assets/GrafanaDashboard.png" width="800">
+
+## Start a fn server and deploy some functions
```
Comment (derekschultz) [inline examples/grafana/README.md]: edit: Start an Fn server...

File: examples/grafana/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+This example requires a Fn server to be running and that you have deployed one or more functions.
```
Comment (derekschultz) [inline examples/grafana/README.md]: edit: requires an Fn server...

File: examples/grafana/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+Now start Prometheus, specifying this config file:
+```
+docker run --name=prometheus -d -p 9090:9090 \
+  --mount type=bind,source=`pwd`/prometheus.yml,target=/etc/prometheus/prometheus.yml \
+  --add-host="localhost:`route | grep default | awk '{print $2}'`" prom/prometheus
+```
+Note: The parameter `` --add-host="localhost:`route | grep default | awk '{print $2}'`" `` means that Prometheus can use localhost to refer to the host. (The expression `` `route | grep default | awk '{print $2}'` ``  returns the IP of the host).
+
+Open a browser on Prometheus's graph tool at [http://localhost:9090/graph](http://localhost:9090/graph). If you wish you can use this to view metrics and display metrics from the fn server: see the [Prometheus](https://prometheus.io/) documentation for instructions. Alternatively continue with the next step to view a ready-made set of graphs in Grafana.
```
Comment (derekschultz) [inline examples/grafana/README.md]: s/fn/Fn/

File: examples/grafana/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+# Here it's the Fn server
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'functions'
+
+    # Override the global default and scrape targets from this job every 5 seconds.
+    scrape_interval: 5s
+
+    static_configs:
+      # Specify all the fn servers from which metrics will be scraped
```
Comment (derekschultz) [inline examples/grafana/README.md]: s/fn/Fn/

**Comment by nigeldeakin** [comment]
Applied all the edits suggested by @derekschultz

**Pull request merged**